File: server/src/main/com/broll/gainea/server/core/cards/impl/play/C_ReplaceGoal.java
package com.broll.gainea.server.core.cards.impl.play;

import com.broll.gainea.server.core.cards.Card;
import com.broll.gainea.server.core.goals.Goal;
import com.broll.gainea.server.core.goals.GoalDifficulty;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class C_ReplaceGoal extends Card {
    public C_ReplaceGoal() {
        super(39, "Zielstrategie", "Wähle aus drei neuen Zielen und ersetze damit ein vorhandenes Ziel");
        setDrawChance(0.5f);
    }

    @Override
    public boolean isPlayable() {
        return true;
    }

    @Override
    protected void play() {
        List<Goal> goals = game.getGoalStorage().getAnyGoals(owner, 3);
        Goal newGoal = goals.get(selectHandler.selectObject("Wähle ein neues Ziel", goals.stream().map(Goal::nt).collect(Collectors.toList())));
        List<Goal> oldGoals = owner.getGoalHandler().getGoals();
        Goal oldGoal = oldGoals.get(selectHandler.selectObject("Welches alte Ziel soll ersetzt werden?", oldGoals.stream().map(Goal::nt).collect(Collectors.toList())));
        owner.getGoalHandler().removeGoal(oldGoal);
        owner.getGoalHandler().newGoal(newGoal);
    }

}

File: server/src/main/com/broll/gainea/server/core/cards/impl/play/C_ReplaceGoal.kt
package com.broll.gainea.server.core.cards.impl.play

import com.broll.gainea.server.core.cards.Card
import com.broll.gainea.server.core.cards.EffectType.OTHER

class C_ReplaceGoal : Card(
    39, OTHER,
    "Zielstrategie",
    "Wähle aus drei neuen Zielen und ersetze damit ein vorhandenes Ziel"
) {
    init {
        drawChance = 0.5f
    }

    override val isPlayable: Boolean
        get() = true

    override fun play() {
        val goals = game.goalStorage.getRandomGoals(owner, 3)
        val newGoal =
            goals[selectHandler.selectObject("Wähle ein neues Ziel", goals.map { it.nt() })]
        val oldGoals = owner.goalHandler.goals
        val oldGoal = oldGoals[selectHandler.selectObject(
            "Welches alte Ziel soll ersetzt werden?",
            oldGoals.map { it.nt() })]
        owner.goalHandler.removeGoal(oldGoal)
        owner.goalHandler.newGoal(newGoal)
    }
}
